session/session.go

```diff
@@ -689,52 +691,55 @@ func (c *cachedTableRenewLease) commitTSCheck(commitTS uint64) bool {
 // If it's not an assertion failure, returns the original error.
 func (s *session) handleAssertionFailure(err error) error {
 	var assertionFailure *tikverr.ErrAssertionFailed
-	if stderrs.As(err, &assertionFailure) {
-		key := assertionFailure.Key
-		newErr := kv.ErrAssertionFailed.GenWithStackByArgs(
-			hex.EncodeToString(key), assertionFailure.String(), assertionFailure.StartTs,
-			assertionFailure.ExistingStartTs, assertionFailure.ExistingCommitTs,
-		)
-
-		var decodeFunc func(kv.Key, *kvrpcpb.MvccGetByKeyResponse, map[string]interface{})
-		// if it's a record key or an index key, decode it
-		if infoSchema, ok := s.sessionVars.TxnCtx.InfoSchema.(infoschema.InfoSchema); ok && infoSchema != nil {
+	if !stderrs.As(err, &assertionFailure) {
+		return err
+	}
+	key := assertionFailure.Key
+	newErr := kv.ErrAssertionFailed.GenWithStackByArgs(
+		hex.EncodeToString(key), assertionFailure.Assertion.String(), assertionFailure.StartTs,
+		assertionFailure.ExistingStartTs, assertionFailure.ExistingCommitTs,
+	)
+
+	if s.GetSessionVars().EnableRedactLog {
+		return newErr
```

Comment:
Hope the transaction event table can be supported someday so we have place to save the details :(